### ISSUE TITLE
Create Firebase Auth Account and Integrate with Hasura Project

### ISSUE THREAD

**Issue by sotoJ24** [safetrustcr/Backend#22]
# Create Firebase Auth Account and Integrate with Hasura Project

Set up Firebase Authentication and integrate it with our Hasura GraphQL API to handle user authentication.

## Problem
We need to:
- Set up Firebase project for authentication
- Configure Hasura to use Firebase JWT tokens
- Set up proper role-based authorization
- Enable multiple sign-in methods (email, Google, wallet)

### 1. Firebase Setup

1. Create new Firebase project:
   - Go to [Firebase Console](https://console.firebase.google.com/)
   - Click "Add project"
   - Name it "SafeTrust-[Environment]" (e.g., SafeTrust-Dev)

2. Enable Authentication methods:
   - Navigate to Authentication > Sign-in method
   - Enable:
     - [ ] Email/Password
     - [x] Google
     - [ ] Custom Authentication (for Web3 wallet)

Follow this tutorial:

https://hasura.io/learn/graphql/hasura-authentication/integrations/firebase/

Just in the roles section, create the roles mentioned below

### 3. Hasura Permission Setup

1. Create roles in Hasura:
   - `anonymous`: For unauthenticated users
   - `user`: For authenticated users
   - `tenant`: For apartment seekers
   - `landlord`: For property owners

2. Configure permissions for `users` table:
```sql
-- Select permissions for user role
{
  "columns": ["id", "email", "last_seen"],
  "filter": {
    "id": { "_eq": "X-Hasura-User-Id" }
  }
}


## Environment Setup

You can use a dummy login page in the frontend or separate to get the JWT token and test it against the current backend.

2. Test Hasura JWT:
```graphql
query TestAuth {
  users {
    id
    email
  }
}
```

Given a valid JWT, this account should

## Prerequisites
- [ ] Firebase account with the most basic plan
- [ ] Hasura project set up

## Success Criteria
- [ ] Firebase project created and configured
- [ ] Authentication methods enabled
- [ ] Hasura JWT configuration working
- [ ] Role-based permissions working
- [ ] Test queries successful with authentication
- [ ] Test this with a dummy login page as the tutorial explains. You can use the frontend repo for this

## Security Checklist
- [ ] JWT secret properly configured
- [ ] Admin secret properly set
- [ ] CORS configured correctly
- [ ] Authentication rules tested
- [ ] Role-based access tested

## Additional Context
This setup will enable:
- Secure user authentication
- Role-based access control
- Multiple sign-in methods
- JWT token validation
- User session management

**Comment by Jonatan-Chaverri** [comment]
I will like to take care of this issue! My name is Jonatan, I'm from Dojo Coding community! I had used firebase auth in the past so I have experience on this area.

**Comment by josephchimebuka** [comment]
Hello I am Joseph I am a frontend developer and blockchain developer and I am also an active contributor here on only dust here is my profile https://app.onlydust.com/u/josephchimebuka. This is my first time to contribute to this repo ill appreciate the opportunity to contribute. this is a couple of frontend and blockchain projects i have built https://metacrypt.vercel.app/ https://mattedsgn.com/
May I try this one?

**Comment by Michaelkingsdev** [comment]
I'd like to take this issue.

**Comment by Unickhub** [comment]
Can I take this issue?

**Comment by ShantelPeters** [comment]
Can I take care of this issue?

**Comment by martinvibes** [comment]
Can I take care of this issue?
i'm a frontend dev and a blockchain dev please kindly assign :)

**Comment by mariocodecr** [comment]
Hii! My name is Mario Araya, I’m a Software Developer with 2+ years of experience. I have worked with backend technologies such as Cairo, Java, and C#, as well as frontend development using React, NextJS, and JavaScript/TypeScript. I’ve made contributions to open-source projects, completed a Starknet Bootcamp, exercises on NodeGuardians, finished Starklings, and participated in multiple hackathons. I’m also a member of the Dojo Coding community here in Costa Rica.

This involves creating a Firebase project, enabling sign-in methods (email/password, Google, Web3 wallet), and configuring Hasura to accept Firebase JWT tokens. Set up roles (anonymous, user, tenant, landlord) in Hasura with role-based access control for different permissions. Test the JWT integration with a dummy login page, ensuring that role-based permissions work correctly. Make sure Firebase and Hasura are properly configured with correct JWT validation and security settings. I have worked on Firebase projects with different clients, so this setup will enhance user management and security in your project.
